hpx/runtime/threads/executors/current_executor.hpp

```diff
@@ -21,11 +21,11 @@ namespace hpx { namespace threads { namespace executors
     namespace detail
     {
         //////////////////////////////////////////////////////////////////////
-        class HPX_EXPORT generic_thread_pool_executor
+        class HPX_EXPORT current_executor
           : public threads::detail::scheduled_executor_base
         {
         public:
-            generic_thread_pool_executor(policies::scheduler_base* scheduler);
+            current_executor(policies::scheduler_base* scheduler);
```

Comment:
It's as the name implies, `current_executor` is a type erased wrapper around the executor the current thread has been scheduled with. `this_thread_executor` can be used to schedule an HPX-thread on the current OS-thread.

The first is nice to keep scheduling HPX-threads on the same executor as the current thread was scheduled with. The second ensures that all HPX-threads scheduled by it will run sequentially wrt each other.